database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/UserViewModel.swift

```diff
@@ -94,6 +114,59 @@ class UserViewModel: ObservableObject {
       NSLog("Error signing out.")
     }
   }
+
+  func post(title: String, body: String) {
+    // check if both title and body are completed
+    if title.isEmpty || body.isEmpty {
+      showAlertMessage(message: "Neither title nor body can be empty.")
+      return
+    }
+
+    // begin loading animation
+    withAnimation {
+      self.isLoading.toggle()
+    }
+
+    let userID = getuid()
+    guard let key = postRef.childByAutoId().key else { return }
+    let post = ["id": key,
```

Comment:
If you follow my suggestion above, the first key-value pair (`"id": key`) can be deleted.